basic_hashtable/b_hashtables.py

```diff
@@ -32,16 +39,45 @@ def hash(string, max):
 # If you are overwriting a value with a different key, print a warning.
 # '''
 def hash_table_insert(hash_table, key, value):
-    pass
-
+    key_hash = hash(key, hash_table.capacity)
+    new_pair = Pair(key, value)
+    if hash_table.storage[key_hash]:
+        print("hash already exists")
+    else:
+        hash_table.storage[key_hash] = new_pair
+
+    # def insert(self, key, value):
+    #     key_hash = _djb2x_hash(key)
+    #     bucket_index = key_hash % self.capacity
+
+    #     new_node = _Node(key, value)
+    #     existing_node = self.bucket_array[bucket_index]
+
+    #     if existing_node:
+    #         last_node = None
+    #         while existing_node:
+    #             if existing_node.key == key:
+    #                 # found existing key, replace value
+    #                 existing_node.value = value
+    #                 return
+    #             last_node = existing_node
+    #             existing_node = existing_node.next_node
+    #         # if we get this far, we didn't find an existing key
+    #         # so just append the new node to the end of the bucket
+    #         last_node.next_node = new_node
+    #     else:
+    #         self.bucket_array[bucket_index] = new_node
 
 # '''
 # Fill this in.
 
 # If you try to remove a value that isn't there, print a warning.
 # '''
+
+
 def hash_table_remove(hash_table, key):
-    pass
+    hash_key = hash(key, hash_table.capacity)
+    hash_table.storage[hash_key] = None
```

Comment:
Don't forget to check the key of the thing you're removing in case there was a collision. You'll want to let the user know.